http/request/client.go

```diff
@@ -3,11 +3,10 @@ package request
 import (
 	"fmt"
 	"net/http"
-
-	"github.com/krateoplatformops/plumbing/endpoints"
 )
 
-func HTTPClientForEndpoint(ep *endpoints.Endpoint) (*http.Client, error) {
+func HTTPClientForEndpoint(opts RequestOptions) (*http.Client, error) {
```

Comment:
I noticed that all functions that used to take an `Endpoint` now take a `RequestOptions` struct instead. For example:

`func HTTPClientForEndpoint(ep *endpoints.Endpoint) …`

became:

`func HTTPClientForEndpoint(opts RequestOptions) …`

But inside the code it looks like the only field actually used is `opts.Endpoint`. If that’s the case, this change might be adding a bit more complexity than needed. Could you share the reasoning behind switching to `RequestOptions` here?